gamemodes/cinema/gamemode/modules/theater/ents/entities/theater_thumbnail/cl_init.lua

```diff
@@ -144,7 +144,7 @@ end
 function ENT:DrawThumbnail()
 
 	-- Thumbnail isn't set yet
-	if self:GetThumbnail() == "" then
+	if self:GetThumbnail() == "" or (self:GetService() == "file" and not GetConVar("swamp_mature_content"):GetBool()) then
```

Comment:
`theater.Services[self:GetService()].Mature` should work (from gamemodes/cinema/gamemode/modules/theater/sh_services.lua, RegisterService())